pallet/deposit/src/lib.rs

```diff
@@ -216,14 +213,13 @@ pub mod pallet {
 			}
 
 			if to_claim.0 != 0 {
-				T::Ring::transfer(&account_id(), who, to_claim.0, AllowDeath)?;
+				T::Ring::transfer(&T::Treasury::get(), who, to_claim.0, AllowDeath)?;
 				Self::deposit_event(Event::DepositsClaimed {
 					owner: who.clone(),
 					deposits: to_claim.1,
 				});
 			}
 			if to_migrate.0 != 0 {
-				T::Ring::transfer(&account_id(), &T::Treasury::get(), to_migrate.0, AllowDeath)?;
 				T::DepositMigrator::migrate(who.clone(), to_migrate.0, to_migrate.2)?;
```

Comment:
I'm thing if we fail with error here, it still end up with side effects like in line 216 (some of the funds might be transferred to user as we are in a loop)

We should emit a DepositMigration Failed events here?

Reply:
There will be an error log on node side.

Not sure if a Substrate event works for us nowadays. Only Darwinia has a Substrate explorer.

Reply:
Let's emit Event anyway, at least we can trace back and index to it.

Reply:
Fortunately, I remember that the value = 0 entry is this user’s first deposit; otherwise, we might have already encountered side effects.